apps/analysis/serializers.py

```diff
@@ -440,6 +440,43 @@ def validate(self, data):
             )
         return data
 
+    def create_or_update_pillar(self, pillar_data, instance):
+        data = {
+            "title": pillar_data.get('title'),
+            "assignee": pillar_data.get('assignee').id,
+            "analysis": instance.id,
+        }
+        pillar_id = pillar_data.get('id', None)
+        if pillar_id:
+            data["id"] = pillar_id
+            analysis_pillar = get_object_or_404(AnalysisPillar, pk=pillar_id)
```

Comment:
Crash when object not found. We need to handle this error. We need to show in serializer error when objects doesnot exists.